rosjava/src/main/java/org/ros/node/topic/TransportHints.java

```diff
@@ -0,0 +1,43 @@
+package org.ros.node.topic;
+
+import java.util.Map;
+
+import org.ros.internal.transport.ConnectionHeaderFields;
+import org.ros.node.ConnectedNode;
+
+import com.google.common.collect.Maps;
+
+
+/**
+ * Provides a way of specifying network transport hints to
+ * {@link ConnectedNode#newSubscriber(String, String)} and
+ * {@link ConnectedNode#newSubscriber(org.ros.namespace.GraphName, String)}.
+ *
+ * @author stefan.glaser@hs-offenburg.de (Stefan Glaser)
+ */
+public class TransportHints {
+
+  Map<String, String> options;
+
+ public TransportHints() {
+   this.options = Maps.newConcurrentMap();
+  }
+
+  public TransportHints(boolean tcpNoDelay) {
+    tcpNoDelay(tcpNoDelay);
```

Comment:
This constructor yields a `NullPointerException` when called, as the default constructor is not called and therefore the map is not created.

Please consider either:
- Calling the default constructor first.
- Moving the initialization of the map to an instance initializer.
- Initialize `options` inline (i.e. `Map<String, String> options = Maps.newConcurrentMap<String, String>();`).

I know in Java 7 it's not necessary to specify `<String, String>` twice, but please do so so as to make it easier to compile `rosjava` for older Java versions.